internal/dkg/state_machine.go

```diff
@@ -802,12 +762,12 @@ func validateEpoch(currentState *DBState, terms *drand.ProposalTerms) error {
 	}
 
 	// aborted or timed out DKGs can be reattempted at the same epoch
-	if terms.Epoch == currentState.Epoch && currentState.State != Aborted && currentState.State != TimedOut {
+	if terms.Epoch == currentState.Epoch && currentState.State != Aborted && currentState.State != TimedOut && currentState.State != Failed {
 		return ErrInvalidEpoch
 	}
 
 	// if we have some leftover state after having left the network, we can accept higher epochs
-	if terms.Epoch > currentState.Epoch+1 && (currentState.State != Left && currentState.State != Fresh && currentState.State != Evicted) {
+	if terms.Epoch > currentState.Epoch+1 && (currentState.State != Left && currentState.State != Fresh) {
```

Comment:
Why is this `> current+1 ` and not `>= current + 1` ?

Reply:
basically, if we are not evicted or leave in an epoch, we must be a `remainer`, therefore epochs should be monotonically increasing. This isn't entirely watertight - if my node is down for the entirety of a DKG process, I'm a `leaver`, but my node doesn't know; in that case, I ought to clear my state or recover somehow.
In the future, we'll ideally gossip (and perhaps sync?) completed DKGs, so that if I'm down for an entire DKG, on startup I can see the newest epoch and realise I'm a leaver